modules/connectors/dhl_express/karrio/providers/dhl_express/rate.py

```diff
@@ -102,7 +102,13 @@ def rate_request(
         raise errors.DestinationNotServicedError(payload.shipper.country_code)
 
     is_document = all([parcel.is_document for parcel in payload.parcels])
-    is_dutiable = is_international and not is_document
+    is_from_EU = payload.shipper.country_code in units.EUCountry
+    is_to_EU = payload.recipient.country_code in units.EUCountry
+    is_dutiable = (
+        is_international
+        and not is_document
+        and not (is_from_EU and is_to_EU)
+    )
```

Comment:
@DarkSwoop what do you think of this. I cleaned it up a bit to make it more readable. 
This checks all the rules you mentioned right
* Exports from EU to the world (non-EU) are dutiable.
* Imports from world (non-EU) to EU are dutiable.
* Imports and exports within EU are not dutiable.

Reply:
@danh91 I like cleaned up code! 😄

Reply:
I am releasing the patch now. 👍🏿